### ISSUE TITLE
Bug when masking spine animation. 

### ISSUE THREAD

**Issue by JacobCaron** [photonstorm/phaser#4454]
## Version
* Phaser Version: 3.16 +
* Operating system: Windows 10
* Browser: All

## Description

**WHAT HAPPENS:** When masking a container and adding a spine animation to it, any additional GameObjects added to the container will not be masked.

**EXPECTATION:** Any subsequent GameObjects added to the masked container will be masked.

## Example Test Code

```
var game = new Phaser.Game({
    type: Phaser.WEBGL, width: 800, height: 600,
    scene: {
        preload: preload,
        pack: { files: [ { type: 'scenePlugin', key: 'SpineWebGLPlugin', url: 'https://labs.phaser.io/plugins/SpineWebGLPlugin.js', sceneKey: 'spine' }]},
        create: create
    }
});

function preload()
{
    this.load.image('logo', 'https://labs.phaser.io/assets/sprites/phaser.png');
    this.load.setPath('https://labs.phaser.io/assets/animations/spine/');
    this.load.spine('boy', 'spineboy.json', 'spineboy.atlas');
}

function create()
{
    var maskedContainer = this.add.container(20, 20);
    var maskShape = this.add.graphics({fillStyle: { color: 0xff0000 }}).fillRectShape(new Phaser.Geom.Rectangle(20, 20, 400, 400));
    maskShape.alpha = .5;
	maskedContainer.setMask(new Phaser.Display.Masks.GeometryMask(this, maskShape));

    var image1 = this.add.image(400, 200, 'logo');
    var image2 = this.add.image(400, 100, 'logo');
    var spine = this.add.spine(200, 400, 'boy', 'run', true);

    maskedContainer.add([image1, spine, image2]); //** BUG: Notice how image2 is not masked. **//
}
```

https://codepen.io/JakeCaron/pen/ErpMKx

**Comment by spayton** [comment]
I've noticed this too.  The rebindPipeline function is called when the spine renderer swaps back to the phaser rendering, it resets the stencil to a certain state, thus losing the stencil you'd like to be kept and applied to subsequent rendering.

It also resets viewport, blendmode and texture binding which may not show as a problem but could cause some unnecessary webgl calls.  This could impact performance for a large number of spine objects.

The simple fix is to have rebindPipeline reinstate the correct stencil.  A better fix would be for rebindPipeline to have parameters to control what does and doesn't get reset, which in this case the spine renderer would not want the stencil reset because it knows it hasn't used it.

**Comment by JacobCaron** [comment]
This is fixed in Phaser 3.17.
